oaknational/Oak-Web-Application
# [CUR-435] Curriculum downloads journey

## Description

- adds the components for the curriculum download journey
- adds some hooks and helpers
  - `usePrerelease` - allow for cookie based pre-release flags
  - `useFetch` - `fetch()` with `OakError`s 
  - `getCookiesWithSchema` - get values from cookies in a type safe way (`zod`)
  - `renderHookWithInitialState` -  allow for testing hooks with the initial state (can I do this another way???)

## Issue(s)

Fixes `CUR-435`

 - Missing `CUR-426` (to be integrated later)
 - Missing `CUR-427` (to be integrated later)

## How to test
Because we want to merge to `main` now we need to put this behind a flag. To add the flag, add the **cookie** `prerelease.curriculum.downloads.enabled=true` (see video)

https://github.com/oaknational/Oak-Web-Application/assets/235915/eca34ecd-b3f6-4968-8acd-f44cd67b3355

You can then head to https://deploy-preview-2399--oak-web-application.netlify.thenational.academy/teachers/curriculum/music-secondary-eduqas/download to test

You should see the following

https://github.com/oaknational/Oak-Web-Application/assets/235915/f2f9cd82-ccdd-4ac3-9eff-6ad9e759ed5f

**Note:** this PR is the UI only and the won't actually download anything, but the UI should function up to that point. Upon submit data is logged to the web inspector console.

Other things to test for

 - No other pages should be effected by this change
 - When the cookie isn't present the page/route should hidden

## Checklist
